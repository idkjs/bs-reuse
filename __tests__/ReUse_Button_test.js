'use strict';

var Jest = require("@glennsl/bs-jest/src/jest.js");
var Curry = require("bs-platform/lib/js/curry.js");
var React = require("react");
var JestDom = require("bs-jest-dom/src/JestDom.bs.js");
var ReUse_Button = require("../src/ReUse_Button.js");
var ReactTestingLibrary = require("bs-react-testing-library/src/ReactTestingLibrary.bs.js");

Jest.describe("getByText", (function (param) {
        Jest.test("Component renders", (function (param) {
                var btn = React.createElement(ReUse_Button.make, {
                      children: "Hello Button"
                    });
                var __x = ReactTestingLibrary.render(undefined, undefined, btn);
                return JestDom.toBeInTheDocument(Jest.Expect.expect(ReactTestingLibrary.getByText(/* `Str */[
                                    4153489,
                                    "Hello Button"
                                  ], undefined, __x)));
              }));
        Jest.test("Component renders", (function (param) {
                var btn = React.createElement(ReUse_Button.make, {
                      type_: "button",
                      children: "Hello Button",
                      variant: /* text */-856044371
                    });
                return Jest.Expect.toMatchSnapshot(Jest.Expect.expect(ReactTestingLibrary.render(undefined, undefined, btn).container));
              }));
        Jest.test("Component renders disabled", (function (param) {
                var onClick = (jest.fn());
                var __x = React.createElement(ReUse_Button.make, {
                      onClick: onClick,
                      type_: "button",
                      children: "Hello Button",
                      disabled: true
                    });
                var btn = ReactTestingLibrary.render(undefined, undefined, __x);
                Curry._2(ReactTestingLibrary.FireEvent.click, undefined, ReactTestingLibrary.getByText(/* `Str */[
                          4153489,
                          "Hello Button"
                        ], undefined, btn));
                ((expect(onClick).not.toHaveBeenCalled()));
                return Jest.pass;
              }));
        return Jest.test("Component onClick prop", (function (param) {
                      var onClick = (jest.fn());
                      var __x = React.createElement(ReUse_Button.make, {
                            onClick: onClick,
                            type_: "button",
                            children: "Hello Button"
                          });
                      var btn = ReactTestingLibrary.render(undefined, undefined, __x);
                      Curry._2(ReactTestingLibrary.FireEvent.click, undefined, ReactTestingLibrary.getByText(/* `Str */[
                                4153489,
                                "Hello Button"
                              ], undefined, btn));
                      ((expect(onClick).toHaveBeenCalled()));
                      return Jest.pass;
                    }));
      }));

/*  Not a pure module */
