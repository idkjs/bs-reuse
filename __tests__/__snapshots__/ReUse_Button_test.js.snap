// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getByText Component renders 1`] = `
<div>
  <button
    class="css-13e6qem css-uzwnr3"
    type="button"
  >
    Hello Button
  </button>
</div>
`;
